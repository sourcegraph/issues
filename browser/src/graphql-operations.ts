import { CampaignState } from '../../shared/src/graphql/schema'
import { OrganizationInvitationResponseType } from '../../shared/src/graphql/schema'
import { EventSource } from '../../shared/src/graphql/schema'
import { RepositoryPermission } from '../../shared/src/graphql/schema'
import { ChangesetState } from '../../shared/src/graphql/schema'
import { ChangesetExternalState } from '../../shared/src/graphql/schema'
import { ChangesetReviewState } from '../../shared/src/graphql/schema'
import { ChangesetCheckState } from '../../shared/src/graphql/schema'
import { SymbolKind } from '../../shared/src/graphql/schema'
import { DiagnosticSeverity } from '../../shared/src/graphql/schema'
import { ExternalServiceKind } from '../../shared/src/graphql/schema'
import { GitRefType } from '../../shared/src/graphql/schema'
import { GitObjectType } from '../../shared/src/graphql/schema'
import { GitRefOrder } from '../../shared/src/graphql/schema'
import { DiffHunkLineType } from '../../shared/src/graphql/schema'
import { LSIFUploadState } from '../../shared/src/graphql/schema'
import { LSIFIndexState } from '../../shared/src/graphql/schema'
import { RepositoryOrderBy } from '../../shared/src/graphql/schema'
import { UserActivePeriod } from '../../shared/src/graphql/schema'
import { SearchVersion } from '../../shared/src/graphql/schema'
import { SearchPatternType } from '../../shared/src/graphql/schema'
import { AlertType } from '../../shared/src/graphql/schema'
import { UserEvent } from '../../shared/src/graphql/schema'
import { ChangesetSpecType } from '../../shared/src/graphql/schema'
export type Maybe<T> = T | null
export type Exact<T extends { [key: string]: any }> = { [K in keyof T]: T[K] }
/* This is an autogenerated file. Do not edit this file directly! */

export interface BrowserGraphQlOperations {
    /** browser/src/shared/backend/diffs.tsx */
    RepositoryComparisonDiff: (variables: RepositoryComparisonDiffVariables) => RepositoryComparisonDiffResult

    /** browser/src/shared/backend/search.tsx */
    SearchSuggestions: (variables: SearchSuggestionsVariables) => SearchSuggestionsResult

    /** browser/src/shared/backend/server.ts */
    SiteProductVersion: (variables: SiteProductVersionVariables) => SiteProductVersionResult

    /** browser/src/shared/backend/userEvents.tsx */
    logUserEvent: (variables: logUserEventVariables) => logUserEventResult

    /** browser/src/shared/backend/userEvents.tsx */
    logEvent: (variables: logEventVariables) => logEventResult

    /** browser/src/shared/code-hosts/phabricator/backend.tsx */
    addPhabricatorRepo: (variables: addPhabricatorRepoVariables) => addPhabricatorRepoResult

    /** browser/src/shared/code-hosts/phabricator/backend.tsx */
    ResolveStagingRev: (variables: ResolveStagingRevVariables) => ResolveStagingRevResult

    /** browser/src/shared/platform/context.test.tsx */
    ResolveRepoTest: (variables: ResolveRepoTestVariables) => ResolveRepoTestResult

    /** browser/src/shared/repo/backend.tsx */
    ResolveRepo: (variables: ResolveRepoVariables) => ResolveRepoResult

    /** browser/src/shared/repo/backend.tsx */
    ResolveRev: (variables: ResolveRevVariables) => ResolveRevResult

    /** browser/src/shared/repo/backend.tsx */
    BlobContent: (variables: BlobContentVariables) => BlobContentResult
}
/** All built-in and custom scalars, mapped to their actual values */
export interface Scalars {
    ID: string
    String: string
    Boolean: boolean
    Int: number
    Float: number
    DateTime: string
    JSONCString: string
    JSONValue: unknown
    GitObjectID: string
}

export interface AddExternalServiceInput {
    kind: ExternalServiceKind
    displayName: Scalars['String']
    config: Scalars['String']
}

export { AlertType }

export { CampaignState }

export { ChangesetCheckState }

export { ChangesetExternalState }

export { ChangesetReviewState }

export { ChangesetSpecType }

export { ChangesetState }

export interface ConfigurationEdit {
    keyPath: Array<KeyPathSegment>
    value?: Maybe<Scalars['JSONValue']>
    valueIsJSONCEncodedString?: Maybe<Scalars['Boolean']>
}

export { DiagnosticSeverity }

export { DiffHunkLineType }

export { EventSource }

export { ExternalServiceKind }

export { GitObjectType }

export { GitRefOrder }

export { GitRefType }

export interface KeyPathSegment {
    property?: Maybe<Scalars['String']>
    index?: Maybe<Scalars['Int']>
}

export { LSIFIndexState }

export { LSIFUploadState }

export interface MarkdownOptions {
    alwaysNil?: Maybe<Scalars['String']>
}

export { OrganizationInvitationResponseType }

export interface ProductLicenseInput {
    tags: Array<Scalars['String']>
    userCount: Scalars['Int']
    expiresAt: Scalars['Int']
}

export interface ProductSubscriptionInput {
    billingPlanID: Scalars['String']
    userCount: Scalars['Int']
}

export { RepositoryOrderBy }

export { RepositoryPermission }

export { SearchPatternType }

export { SearchVersion }

export interface SettingsEdit {
    keyPath: Array<KeyPathSegment>
    value?: Maybe<Scalars['JSONValue']>
    valueIsJSONCEncodedString?: Maybe<Scalars['Boolean']>
}

export interface SettingsMutationGroupInput {
    subject: Scalars['ID']
    lastID?: Maybe<Scalars['Int']>
}

export interface SurveySubmissionInput {
    email?: Maybe<Scalars['String']>
    score: Scalars['Int']
    reason?: Maybe<Scalars['String']>
    better?: Maybe<Scalars['String']>
}

export { SymbolKind }

export interface UpdateExternalServiceInput {
    id: Scalars['ID']
    displayName?: Maybe<Scalars['String']>
    config?: Maybe<Scalars['String']>
}

export { UserActivePeriod }

export { UserEvent }

export interface UserPermission {
    bindID: Scalars['String']
    permission?: Maybe<RepositoryPermission>
}

export type RepositoryComparisonDiffVariables = Exact<{
    repo: Scalars['String']
    base: Maybe<Scalars['String']>
    head: Maybe<Scalars['String']>
    first: Maybe<Scalars['Int']>
}>

export type RepositoryComparisonDiffResult = {
    repository: Maybe<{ comparison: { fileDiffs: { totalCount: Maybe<number>; nodes: Array<FileDiffFields> } } }>
}

export type FileDiffFields = { oldPath: Maybe<string>; newPath: Maybe<string>; internalID: string }

export type SymbolFields = {
    __typename: 'Symbol'
    name: string
    containerName: Maybe<string>
    url: string
    kind: SymbolKind
    location: { url: string; resource: { path: string; repository: { name: string } } }
}

export type SearchSuggestionsVariables = Exact<{
    query: Scalars['String']
    first: Scalars['Int']
}>

export type SearchSuggestionsResult = {
    search: Maybe<{
        suggestions: Array<
            | { __typename: 'Repository'; name: string; url: string }
            | {
                  __typename: 'File'
                  path: string
                  name: string
                  isDirectory: boolean
                  url: string
                  repository: { name: string }
              }
            | SymbolFields
        >
    }>
}

export type SiteProductVersionVariables = Exact<{ [key: string]: never }>

export type SiteProductVersionResult = {
    site: { productVersion: string; buildVersion: string; hasCodeIntelligence: boolean }
}

export type logUserEventVariables = Exact<{
    event: UserEvent
    userCookieID: Scalars['String']
}>

export type logUserEventResult = { logUserEvent: Maybe<{ alwaysNil: Maybe<string> }> }

export type logEventVariables = Exact<{
    name: Scalars['String']
    userCookieID: Scalars['String']
    url: Scalars['String']
    source: EventSource
    argument: Maybe<Scalars['String']>
}>

export type logEventResult = { logEvent: Maybe<{ alwaysNil: Maybe<string> }> }

export type addPhabricatorRepoVariables = Exact<{
    callsign: Scalars['String']
    repoName: Scalars['String']
    phabricatorURL: Scalars['String']
}>

export type addPhabricatorRepoResult = { addPhabricatorRepo: Maybe<{ alwaysNil: Maybe<string> }> }

export type ResolveStagingRevVariables = Exact<{
    repoName: Scalars['String']
    diffID: Scalars['ID']
    baseRev: Scalars['String']
    patch: Maybe<Scalars['String']>
    date: Maybe<Scalars['String']>
    authorName: Maybe<Scalars['String']>
    authorEmail: Maybe<Scalars['String']>
    description: Maybe<Scalars['String']>
}>

export type ResolveStagingRevResult = { resolvePhabricatorDiff: Maybe<{ oid: string }> }

export type ResolveRepoTestVariables = Exact<{
    repoName: Scalars['String']
}>

export type ResolveRepoTestResult = { repository: Maybe<{ url: string }> }

export type ResolveRepoVariables = Exact<{
    rawRepoName: Scalars['String']
}>

export type ResolveRepoResult = { repository: Maybe<{ name: string }> }

export type ResolveRevVariables = Exact<{
    repoName: Scalars['String']
    revision: Scalars['String']
}>

export type ResolveRevResult = {
    repository: Maybe<{ mirrorInfo: { cloned: boolean }; commit: Maybe<{ oid: string }> }>
}

export type BlobContentVariables = Exact<{
    repoName: Scalars['String']
    commitID: Scalars['String']
    filePath: Scalars['String']
}>

export type BlobContentResult = {
    repository: Maybe<{ commit: Maybe<{ file: Maybe<{ content: string } | { content: string }> }> }>
}
