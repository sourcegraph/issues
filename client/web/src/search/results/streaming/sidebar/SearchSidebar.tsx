import { VersionContextProps } from '@sourcegraph/shared/src/search/util'
import React from 'react'
import { CaseSensitivityProps, PatternTypeProps, SearchContextProps } from '../../..'

import styles from './SearchSidebar.module.scss'
import { SidebarResultTypeSection } from './SidebarResultTypeSection'

const SearchSidebarSection: React.FunctionComponent<{ header: string }> = ({ header, children }) => (
    <div>
        <h5>{header}</h5>
        <div>{children}</div>
    </div>
)

export interface SearchSidebarProps
    extends Omit<PatternTypeProps, 'setPatternType'>,
        Omit<CaseSensitivityProps, 'setCaseSensitivity'>,
        VersionContextProps,
        Pick<SearchContextProps, 'selectedSearchContextSpec'> {
    query: string
}

export const SearchSidebar: React.FunctionComponent<SearchSidebarProps> = props => (
    <div className={styles.searchSidebar}>
        <SearchSidebarSection header="Result types">
            <SidebarResultTypeSection {...props} />
        </SearchSidebarSection>
        <SearchSidebarSection header="Dynamic filters" />
        <SearchSidebarSection header="Repositories" />
        <SearchSidebarSection header="Search snippets" />
        <SearchSidebarSection header="Quicklinks" />
    </div>
)
