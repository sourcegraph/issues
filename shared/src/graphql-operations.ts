import { CampaignState } from './graphql/schema'
import { OrganizationInvitationResponseType } from './graphql/schema'
import { EventSource } from './graphql/schema'
import { RepositoryPermission } from './graphql/schema'
import { ChangesetState } from './graphql/schema'
import { ChangesetExternalState } from './graphql/schema'
import { ChangesetReviewState } from './graphql/schema'
import { ChangesetCheckState } from './graphql/schema'
import { SymbolKind } from './graphql/schema'
import { DiagnosticSeverity } from './graphql/schema'
import { ExternalServiceKind } from './graphql/schema'
import { GitRefType } from './graphql/schema'
import { GitObjectType } from './graphql/schema'
import { GitRefOrder } from './graphql/schema'
import { DiffHunkLineType } from './graphql/schema'
import { LSIFUploadState } from './graphql/schema'
import { LSIFIndexState } from './graphql/schema'
import { RepositoryOrderBy } from './graphql/schema'
import { UserActivePeriod } from './graphql/schema'
import { SearchVersion } from './graphql/schema'
import { SearchPatternType } from './graphql/schema'
import { AlertType } from './graphql/schema'
import { UserEvent } from './graphql/schema'
import { ChangesetSpecType } from './graphql/schema'
export type Maybe<T> = T | null
export type Exact<T extends { [key: string]: any }> = { [K in keyof T]: T[K] }
/* This is an autogenerated file. Do not edit this file directly! */

export interface SharedGraphQlOperations {
    /** shared/src/backend/repo.ts */
    ResolveRawRepoName: (variables: ResolveRawRepoNameVariables) => ResolveRawRepoNameResult

    /** shared/src/extensions/helpers.ts */
    Extensions: (variables: ExtensionsVariables) => ExtensionsResult

    /** shared/src/settings/edit.ts */
    ViewerSettings: (variables: ViewerSettingsVariables) => ViewerSettingsResult

    /** shared/src/settings/edit.ts */
    EditSettings: (variables: EditSettingsVariables) => EditSettingsResult

    /** shared/src/settings/edit.ts */
    OverwriteSettings: (variables: OverwriteSettingsVariables) => OverwriteSettingsResult
}
/** All built-in and custom scalars, mapped to their actual values */
export interface Scalars {
    ID: string
    String: string
    Boolean: boolean
    Int: number
    Float: number
    DateTime: string
    JSONCString: string
    JSONValue: unknown
    GitObjectID: string
}

export interface AddExternalServiceInput {
    kind: ExternalServiceKind
    displayName: Scalars['String']
    config: Scalars['String']
}

export { AlertType }

export { CampaignState }

export { ChangesetCheckState }

export { ChangesetExternalState }

export { ChangesetReviewState }

export { ChangesetSpecType }

export { ChangesetState }

export interface ConfigurationEdit {
    keyPath: Array<KeyPathSegment>
    value?: Maybe<Scalars['JSONValue']>
    valueIsJSONCEncodedString?: Maybe<Scalars['Boolean']>
}

export { DiagnosticSeverity }

export { DiffHunkLineType }

export { EventSource }

export { ExternalServiceKind }

export { GitObjectType }

export { GitRefOrder }

export { GitRefType }

export interface KeyPathSegment {
    property?: Maybe<Scalars['String']>
    index?: Maybe<Scalars['Int']>
}

export { LSIFIndexState }

export { LSIFUploadState }

export interface MarkdownOptions {
    alwaysNil?: Maybe<Scalars['String']>
}

export { OrganizationInvitationResponseType }

export interface ProductLicenseInput {
    tags: Array<Scalars['String']>
    userCount: Scalars['Int']
    expiresAt: Scalars['Int']
}

export interface ProductSubscriptionInput {
    billingPlanID: Scalars['String']
    userCount: Scalars['Int']
}

export { RepositoryOrderBy }

export { RepositoryPermission }

export { SearchPatternType }

export { SearchVersion }

export interface SettingsEdit {
    keyPath: Array<KeyPathSegment>
    value?: Maybe<Scalars['JSONValue']>
    valueIsJSONCEncodedString?: Maybe<Scalars['Boolean']>
}

export interface SettingsMutationGroupInput {
    subject: Scalars['ID']
    lastID?: Maybe<Scalars['Int']>
}

export interface SurveySubmissionInput {
    email?: Maybe<Scalars['String']>
    score: Scalars['Int']
    reason?: Maybe<Scalars['String']>
    better?: Maybe<Scalars['String']>
}

export { SymbolKind }

export interface UpdateExternalServiceInput {
    id: Scalars['ID']
    displayName?: Maybe<Scalars['String']>
    config?: Maybe<Scalars['String']>
}

export { UserActivePeriod }

export { UserEvent }

export interface UserPermission {
    bindID: Scalars['String']
    permission?: Maybe<RepositoryPermission>
}

export type ResolveRawRepoNameVariables = Exact<{
    repoName: Scalars['String']
}>

export type ResolveRawRepoNameResult = { repository: Maybe<{ uri: string; mirrorInfo: { cloned: boolean } }> }

export type ExtensionsVariables = Exact<{
    first: Scalars['Int']
    prioritizeExtensionIDs: Array<Scalars['String']>
}>

export type ExtensionsResult = {
    extensionRegistry: {
        extensions: {
            nodes: Array<{
                id: string
                extensionID: string
                url: string
                viewerCanAdminister: boolean
                manifest: Maybe<{ raw: string }>
            }>
        }
    }
}

export type SettingsCascadeFields = {
    final: string
    subjects: Array<
        | {
              __typename: 'User'
              username: string
              displayName: Maybe<string>
              id: string
              settingsURL: Maybe<string>
              viewerCanAdminister: boolean
              latestSettings: Maybe<{ id: number; contents: string }>
          }
        | {
              __typename: 'Org'
              name: string
              displayName: Maybe<string>
              id: string
              settingsURL: Maybe<string>
              viewerCanAdminister: boolean
              latestSettings: Maybe<{ id: number; contents: string }>
          }
        | {
              __typename: 'Site'
              siteID: string
              id: string
              settingsURL: Maybe<string>
              viewerCanAdminister: boolean
              latestSettings: Maybe<{ id: number; contents: string }>
          }
        | {
              __typename: 'DefaultSettings'
              id: string
              settingsURL: Maybe<string>
              viewerCanAdminister: boolean
              latestSettings: Maybe<{ id: number; contents: string }>
          }
    >
}

export type ViewerSettingsVariables = Exact<{ [key: string]: never }>

export type ViewerSettingsResult = { viewerSettings: SettingsCascadeFields }

export type EditSettingsVariables = Exact<{
    subject: Scalars['ID']
    lastID: Maybe<Scalars['Int']>
    edit: ConfigurationEdit
}>

export type EditSettingsResult = {
    configurationMutation: Maybe<{ editConfiguration: Maybe<{ empty: Maybe<{ alwaysNil: Maybe<string> }> }> }>
}

export type OverwriteSettingsVariables = Exact<{
    subject: Scalars['ID']
    lastID: Maybe<Scalars['Int']>
    contents: Scalars['String']
}>

export type OverwriteSettingsResult = {
    settingsMutation: Maybe<{ overwriteSettings: Maybe<{ empty: Maybe<{ alwaysNil: Maybe<string> }> }> }>
}
