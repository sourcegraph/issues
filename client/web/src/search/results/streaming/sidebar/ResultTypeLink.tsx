import React from 'react'

import { Link } from '@sourcegraph/shared/src/components/Link'
import { buildSearchURLQuery } from '@sourcegraph/shared/src/util/url'

import { toggleSearchType } from '../../../helpers'
import { SearchType } from '../../SearchResults'

import { SearchSidebarProps } from './SearchSidebar'
import { SidebarSectionItem } from './SearchSidebarSection'
import styles from './SearchSidebarSection.module.scss'

interface ResultTypeLinkProps extends SearchSidebarProps {
    type: SearchType
}

const ResultTypeLink: React.FunctionComponent<ResultTypeLinkProps> = ({
    type,
    query,
    patternType,
    caseSensitive,
    versionContext,
    selectedSearchContextSpec,
}) => {
    const typeToggledQuery = toggleSearchType(query, type)
    const builtURLQuery = buildSearchURLQuery(
        typeToggledQuery,
        patternType,
        caseSensitive,
        versionContext,
        selectedSearchContextSpec
    )

    return (
        <Link to={{ pathname: '/search', search: builtURLQuery }} className={styles.sidebarSectionListItem}>
            <span className="text-monospace search-query-link">
                <span className="search-filter-keyword">type:</span>
                {type}
            </span>
        </Link>
    )
}

export const getResultTypeLinks = (props: SearchSidebarProps): SidebarSectionItem[] => {
    const types: Exclude<SearchType, null>[] = ['file', 'repo', 'path', 'symbol', 'diff', 'commit']

    return types.map(type => ({ key: type, node: <ResultTypeLink {...props} type={type} /> }))
}
