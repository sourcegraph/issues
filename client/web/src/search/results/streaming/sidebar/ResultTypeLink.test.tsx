import { mount } from 'enzyme'
import React from 'react'

import { SearchPatternType } from '../../../../graphql-operations'

import { getResultTypeLinks } from './ResultTypeLink'
import { SearchSidebarProps } from './SearchSidebar'

const defaultProps: SearchSidebarProps = {
    caseSensitive: false,
    patternType: SearchPatternType.literal,
    versionContext: undefined,
    selectedSearchContextSpec: 'global',
    query: 'test',
}

describe('SidebarResultTypeSection', () => {
    it('should have correct links when type not apresent', () => {
        const links = getResultTypeLinks(defaultProps)
        expect(
            mount(
                <>
                    {links.map(link => (
                        <span key={link.key}>{link.node}</span>
                    ))}
                </>
            )
        ).toMatchSnapshot()
    })

    it('should have correct links when type already exists in query', () => {
        const links = getResultTypeLinks({ ...defaultProps, query: 'test type:repo' })
        expect(
            mount(
                <>
                    {links.map(link => (
                        <span key={link.key}>{link.node}</span>
                    ))}
                </>
            )
        ).toMatchSnapshot()
    })
})
