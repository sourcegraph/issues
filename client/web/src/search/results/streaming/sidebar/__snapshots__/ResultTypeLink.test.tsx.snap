// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarResultTypeSection should have correct links when type already exists in query 1`] = `
Array [
  <span
    key="file"
  >
    <ResultTypeLink
      caseSensitive={false}
      patternType="literal"
      query="test type:repo"
      selectedSearchContextSpec="global"
      type="file"
    >
      <AnchorLink
        className="sidebarSectionListItem"
        to={
          Object {
            "pathname": "/search",
            "search": "q=context:global+test+type:file&patternType=literal",
          }
        }
      >
        <a
          className="sidebarSectionListItem"
          href="/search?q=context:global+test+type:file&patternType=literal"
        >
          <span
            className="text-monospace search-query-link"
          >
            <span
              className="search-filter-keyword"
            >
              type:
            </span>
            file
          </span>
        </a>
      </AnchorLink>
    </ResultTypeLink>
  </span>,
  <span
    key="repo"
  >
    <ResultTypeLink
      caseSensitive={false}
      patternType="literal"
      query="test type:repo"
      selectedSearchContextSpec="global"
      type="repo"
    >
      <AnchorLink
        className="sidebarSectionListItem"
        to={
          Object {
            "pathname": "/search",
            "search": "q=context:global+test+type:repo&patternType=literal",
          }
        }
      >
        <a
          className="sidebarSectionListItem"
          href="/search?q=context:global+test+type:repo&patternType=literal"
        >
          <span
            className="text-monospace search-query-link"
          >
            <span
              className="search-filter-keyword"
            >
              type:
            </span>
            repo
          </span>
        </a>
      </AnchorLink>
    </ResultTypeLink>
  </span>,
  <span
    key="path"
  >
    <ResultTypeLink
      caseSensitive={false}
      patternType="literal"
      query="test type:repo"
      selectedSearchContextSpec="global"
      type="path"
    >
      <AnchorLink
        className="sidebarSectionListItem"
        to={
          Object {
            "pathname": "/search",
            "search": "q=context:global+test+type:path&patternType=literal",
          }
        }
      >
        <a
          className="sidebarSectionListItem"
          href="/search?q=context:global+test+type:path&patternType=literal"
        >
          <span
            className="text-monospace search-query-link"
          >
            <span
              className="search-filter-keyword"
            >
              type:
            </span>
            path
          </span>
        </a>
      </AnchorLink>
    </ResultTypeLink>
  </span>,
  <span
    key="symbol"
  >
    <ResultTypeLink
      caseSensitive={false}
      patternType="literal"
      query="test type:repo"
      selectedSearchContextSpec="global"
      type="symbol"
    >
      <AnchorLink
        className="sidebarSectionListItem"
        to={
          Object {
            "pathname": "/search",
            "search": "q=context:global+test+type:symbol&patternType=literal",
          }
        }
      >
        <a
          className="sidebarSectionListItem"
          href="/search?q=context:global+test+type:symbol&patternType=literal"
        >
          <span
            className="text-monospace search-query-link"
          >
            <span
              className="search-filter-keyword"
            >
              type:
            </span>
            symbol
          </span>
        </a>
      </AnchorLink>
    </ResultTypeLink>
  </span>,
  <span
    key="diff"
  >
    <ResultTypeLink
      caseSensitive={false}
      patternType="literal"
      query="test type:repo"
      selectedSearchContextSpec="global"
      type="diff"
    >
      <AnchorLink
        className="sidebarSectionListItem"
        to={
          Object {
            "pathname": "/search",
            "search": "q=context:global+test+type:diff&patternType=literal",
          }
        }
      >
        <a
          className="sidebarSectionListItem"
          href="/search?q=context:global+test+type:diff&patternType=literal"
        >
          <span
            className="text-monospace search-query-link"
          >
            <span
              className="search-filter-keyword"
            >
              type:
            </span>
            diff
          </span>
        </a>
      </AnchorLink>
    </ResultTypeLink>
  </span>,
  <span
    key="commit"
  >
    <ResultTypeLink
      caseSensitive={false}
      patternType="literal"
      query="test type:repo"
      selectedSearchContextSpec="global"
      type="commit"
    >
      <AnchorLink
        className="sidebarSectionListItem"
        to={
          Object {
            "pathname": "/search",
            "search": "q=context:global+test+type:commit&patternType=literal",
          }
        }
      >
        <a
          className="sidebarSectionListItem"
          href="/search?q=context:global+test+type:commit&patternType=literal"
        >
          <span
            className="text-monospace search-query-link"
          >
            <span
              className="search-filter-keyword"
            >
              type:
            </span>
            commit
          </span>
        </a>
      </AnchorLink>
    </ResultTypeLink>
  </span>,
]
`;

exports[`SidebarResultTypeSection should have correct links when type not apresent 1`] = `
Array [
  <span
    key="file"
  >
    <ResultTypeLink
      caseSensitive={false}
      patternType="literal"
      query="test"
      selectedSearchContextSpec="global"
      type="file"
    >
      <AnchorLink
        className="sidebarSectionListItem"
        to={
          Object {
            "pathname": "/search",
            "search": "q=context:global+test+type:file&patternType=literal",
          }
        }
      >
        <a
          className="sidebarSectionListItem"
          href="/search?q=context:global+test+type:file&patternType=literal"
        >
          <span
            className="text-monospace search-query-link"
          >
            <span
              className="search-filter-keyword"
            >
              type:
            </span>
            file
          </span>
        </a>
      </AnchorLink>
    </ResultTypeLink>
  </span>,
  <span
    key="repo"
  >
    <ResultTypeLink
      caseSensitive={false}
      patternType="literal"
      query="test"
      selectedSearchContextSpec="global"
      type="repo"
    >
      <AnchorLink
        className="sidebarSectionListItem"
        to={
          Object {
            "pathname": "/search",
            "search": "q=context:global+test+type:repo&patternType=literal",
          }
        }
      >
        <a
          className="sidebarSectionListItem"
          href="/search?q=context:global+test+type:repo&patternType=literal"
        >
          <span
            className="text-monospace search-query-link"
          >
            <span
              className="search-filter-keyword"
            >
              type:
            </span>
            repo
          </span>
        </a>
      </AnchorLink>
    </ResultTypeLink>
  </span>,
  <span
    key="path"
  >
    <ResultTypeLink
      caseSensitive={false}
      patternType="literal"
      query="test"
      selectedSearchContextSpec="global"
      type="path"
    >
      <AnchorLink
        className="sidebarSectionListItem"
        to={
          Object {
            "pathname": "/search",
            "search": "q=context:global+test+type:path&patternType=literal",
          }
        }
      >
        <a
          className="sidebarSectionListItem"
          href="/search?q=context:global+test+type:path&patternType=literal"
        >
          <span
            className="text-monospace search-query-link"
          >
            <span
              className="search-filter-keyword"
            >
              type:
            </span>
            path
          </span>
        </a>
      </AnchorLink>
    </ResultTypeLink>
  </span>,
  <span
    key="symbol"
  >
    <ResultTypeLink
      caseSensitive={false}
      patternType="literal"
      query="test"
      selectedSearchContextSpec="global"
      type="symbol"
    >
      <AnchorLink
        className="sidebarSectionListItem"
        to={
          Object {
            "pathname": "/search",
            "search": "q=context:global+test+type:symbol&patternType=literal",
          }
        }
      >
        <a
          className="sidebarSectionListItem"
          href="/search?q=context:global+test+type:symbol&patternType=literal"
        >
          <span
            className="text-monospace search-query-link"
          >
            <span
              className="search-filter-keyword"
            >
              type:
            </span>
            symbol
          </span>
        </a>
      </AnchorLink>
    </ResultTypeLink>
  </span>,
  <span
    key="diff"
  >
    <ResultTypeLink
      caseSensitive={false}
      patternType="literal"
      query="test"
      selectedSearchContextSpec="global"
      type="diff"
    >
      <AnchorLink
        className="sidebarSectionListItem"
        to={
          Object {
            "pathname": "/search",
            "search": "q=context:global+test+type:diff&patternType=literal",
          }
        }
      >
        <a
          className="sidebarSectionListItem"
          href="/search?q=context:global+test+type:diff&patternType=literal"
        >
          <span
            className="text-monospace search-query-link"
          >
            <span
              className="search-filter-keyword"
            >
              type:
            </span>
            diff
          </span>
        </a>
      </AnchorLink>
    </ResultTypeLink>
  </span>,
  <span
    key="commit"
  >
    <ResultTypeLink
      caseSensitive={false}
      patternType="literal"
      query="test"
      selectedSearchContextSpec="global"
      type="commit"
    >
      <AnchorLink
        className="sidebarSectionListItem"
        to={
          Object {
            "pathname": "/search",
            "search": "q=context:global+test+type:commit&patternType=literal",
          }
        }
      >
        <a
          className="sidebarSectionListItem"
          href="/search?q=context:global+test+type:commit&patternType=literal"
        >
          <span
            className="text-monospace search-query-link"
          >
            <span
              className="search-filter-keyword"
            >
              type:
            </span>
            commit
          </span>
        </a>
      </AnchorLink>
    </ResultTypeLink>
  </span>,
]
`;
